### ISSUE TITLE
Workflow reference resource

### ISSUE THREAD

**Issue by kenazk** [lyraproj/lyra#115]
**Is your feature request related to a problem? Please describe.**
I would like to be able to describe Lyra workflows as resources within a workflow so that I can string together several workflows in a bigger workflow. Think GitHub actions like experience. 

**Describe the solution you'd like**
A Lyra provider and Workflow resource that I can use to create a `main.yaml` workflow that contains a bunch of workflows underneath it.

**Comment by kenazk** [comment]
@thallgren did you already have something in my mind for something like this?

**Comment by thallgren** [comment]
@kenazk you should be able to nest workflows already. Did you try it? A workflow contains activities. An activity can be a resource or a workflow in Yaml.

**Comment by kenazk** [comment]
I tried the following example. `aws_vpc_yaml` is another workflow defined in another file in `plugins` dir.  

```
aws_vpc_instance_tf_yaml:
  typespace: TerraformAws
  input:
    tags:
      type: Hash[String,String]
      lookup: aws.tags
  output:
    aws_vpc_id: String
    aws_subnet_id: String
  activities:
    aws_vpc_yaml:
      input: $tags
      output: $aws_subnet_id 
    aws_instance:
      input: $aws_subnet_id
      state:
        tags: $tags
        instance_type: 't2.nano'
        ami: 'ami-f90a4880'
        subnet_id: $aws_subnet_id
```

Ran into the following:
```
panic: rpc error: code = Unknown desc = block may only contain workflow activities. %!{actual}(MISSING) is not supported here  [recovered]
	panic: rpc error: code = Unknown desc = block may only contain workflow activities. %!{actual}(MISSING) is not supported here
```

**Comment by kenazk** [comment]
Ok, sorry for the confusion. 

What I'm looking for is the ability to _reference_ an external activity that is defined in another file. This would then allow me to define a `main.yaml` workflow that could be something like:

```
main:
  typespace:  aws
  input:
     tags:
        type: Hash[String,String]
        lookup: aws.tags
  output:
    aws_vpc_id: String
    aws_subnet_id: String
    aws_instance_id: String
  activities:
     # This is a workflow that creates the VPC and subnet 
     aws_vpc:
         input: $tags
         output: $aws_subnet_id 
     # This is a workflow that creates the instance based on the subnet_id
     aws_instance:
         input: $aws_subnet_id
         output: $aws_instance_id
```

Per @thallgren:
> The objective is to design a reference to an external activity (not necessarily a workflow, may be an externally defined resource or action). This is related to file based loader, packaging, and directory structure.
> At present, you may define a nested embedded workflow, but reference is yet to be designed and implemented.

**Comment by thallgren** [comment]
Important features when referencing an external activity:

- The name of the actual reference is not necessarily the same as the name of the referenced activity.
- Input and output names might differ from input and output names in the referenced activity.

I think we need a YAML syntax similar to:

    aws_vpc:
      input:
        tags:
          type: Hash[String, String]
          refName: tagEntries
      output:
        awsSubnetId:
          type: String
          refName: subnetId
      reference: aws_vpc

All entries except `reference` would be optional and default to the name, input, and output of the referenced entry. The `reference` key must however be present to distinguish the entry from a workflow (has mandatory `activities`) and a resource  (has mandatory `state`), or an activity that is incorrectly specified (forgotten `state` for instance).

At minimum, a reference using all defaults would look like this:

    aws_vpc:
      reference:

No input and output needed since they are specified by the referenced activity and not overridden here. we could even allow:

    aws_vpc: reference

**Comment by ahpook** [comment]
I think this is closed by #26 ?
